crux-core/src/crux/kv/index_store.clj

```diff
@@ -193,6 +193,17 @@
             value (key-suffix k (+ c/index-id-size eid-size c/id-size c/id-size))]
         (->Quad attr entity content-hash value)))))
 
+(defn- decode-ecav-value-from ^ints [^DirectBuffer v]
+  (let [v-size (.capacity v)]
+    (when-not (zero? v-size)
+      (let [len (/ v-size Integer/BYTES)
+            out (int-array len)]
+        (dotimes [idx len]
+          (aset out idx (.getInt v (* idx Integer/BYTES))))
+        out))))
+
+(defrecord ECAVEntry [a v idx])
```

Comment:
Might want to be a deftype with an actual `int` idx, see next comment.

Reply:
Did give this a go, but can't just make this an `int` because it's nullable in the case where the value isn't a collection. Could use `-2` instead, say, if it'd make a reasonable difference, but I'm not sure there's much we can do about passing it to `sort-by` fairly soon afterwards?